x/dsmr/block.go

```diff
@@ -82,6 +86,32 @@ func signChunk[T Tx](
 	return newChunk(chunk, pkBytes, signature)
 }
 
+func (c *Chunk[T]) Verify(networkID uint32, chainID ids.ID) error {
+	signature, err := bls.SignatureFromBytes(c.Signature[:])
+	if err != nil {
+		return err
+	}
+	pk, err := bls.PublicKeyFromCompressedBytes(c.Signer[:])
+	if err != nil {
+		return err
+	}
+	packer := wrappers.Packer{Bytes: make([]byte, 0, InitialChunkSize), MaxSize: consts.NetworkSizeLimit}
+	if err := codec.LinearCodec.MarshalInto(c.UnsignedChunk, &packer); err != nil {
+		return err
+	}
+
+	msg, err := warp.NewUnsignedMessage(networkID, chainID, packer.Bytes)
```

Comment:
can we use `c.bytes` rather than re-packing the message here?

Reply:
We need to use warp signature verification here to factor in the current validator set as well.